lab7/src/main/java/edu/csula/storage/mysql/EventsDAOImpl.java

```
@@ -0,0 +1,106 @@
+			while (rs.next()) {
+				int id = rs.getInt(1);
+				String name = rs.getString(2);
+				String description = rs.getString(3);
+				int triggerAt = rs.getInt(4);
+				events.add(new Event(id, name, description, triggerAt));
+			}
+		} catch (SQLException e) {
+			e.printStackTrace();
+			return new ArrayList<>();
+		}
+		return events;
+	}
+
+	@Override
+	public Optional<Event> getById(int id) {
+		try (Connection c = context.getConnection(); PreparedStatement  stmt = c.prepareStatement(getByIdQuery)) {
+			
+
+			stmt.setInt(1, id);
+
+			ResultSet rs =  stmt.executeQuery();
+			if (rs.next()) {
+				int i = rs.getInt(1);
+				String name = rs.getString(2);
+				String desc = rs.getString(3);
+				int trigger = rs.getInt(4);
+				Event event = new Event(id, name, desc, trigger);
+				return Optional.of(event);
+			}
+
+		} catch (SQLException e) {
+			e.printStackTrace();
+			Optional.empty();
+		}
+		return Optional.empty();
+	}
+
+	@Override
+	public void set(int id, Event event) {
+		try (Connection c = context.getConnection(); PreparedStatement stmt = c.prepareStatement(setQuery)){
+			stmt.setInt(1, id);
+			stmt.setString(2, event.getName());
+			stmt.setString(3, event.getDescription());
+			stmt.setInt(4, event.getTriggerAt());
+			stmt.setInt(5, event.getId());
+			stmt.executeUpdate();
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+	}
+
+	@Override
+	public void add(Event event) {
+		try (Connection c = context.getConnection(); PreparedStatement stmt = c.prepareStatement(addQuery)) {
+			stmt.setString(1, event.getName());
+			stmt.setString(2, event.getDescription());
+			stmt.setInt(3, event.getTriggerAt());
+			stmt.executeUpdate();
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+
+
+	}
+
+	@Override
+	public void remove(int id) {
+		try(Connection c = context.getConnection(); PreparedStatement  stmt = c.prepareStatement(removeQuery)){
+			stmt.setInt(1, id);
+			stmt.executeUpdate();
+		}catch(SQLException e) {
+			
```

should probably still print out the exception if any. Empty catch block is evil.